### PR TITLE
Add .travis.yml

### PR DESCRIPTION
**Pull request by aried3r**
This probably won't work out of the box, but should be a good starting point.

Gemfiles/Ruby combinations seem to work:
screen shot 2017-03-14 at 12 35 23

### CODE REVIEW

**Comment by aried3r** [comment]
The tests now run via PostgreSQL but they fail. It seems many tests are run with database models although they are [only added for SQLite?](https://github.com/expectedbehavior/acts_as_archival/blob/d65056281dc741a7f73dee18515a2da4b90852c5/test/schema.rb#L27)

**Comment by aried3r** [comment]
Changed it to SQLite, in case the tests then work.

**Comment by janxious** [comment]
Basically there are different tests for the transactional systems under test.  You can see that at work in `test/transaction_test.rb`. It may not be necessary to differentiate any more. This project was created before sqlite had transactions and I think it's possibly a relic of that.

Said another way, we could probably strip out all the database-specific pieces now because they were valuable 4 or 5y ago but now are probably just getting in the way.

**Comment by aried3r** [comment]
I think I get what you're saying, but I don't yet quite grasp what is going on in `transaction_test.rb`. Do you think you could help me and apply the necessary changed in this branch? Maybe we can get a successful SQLite run after your changes.

I guess the `PgArchival` and `MysqlArchival` classes will go, which would make #29 obsolete.

**Comment by janxious** [comment]
> I don't yet quite grasp what is going on in `transaction_test.rb`

It basically is the test that will connect to mysql/postgres to make sure data is not being persisted in the case of exception during archive/unarchive. I don't think we need to be specific any more as regards the db engine as for the purposes of this gem's functioning, mysql, postgres, and sqlite have feature parity around transactions (we are not using checkpoints or sub-transactions).

So… yeah, we can probably just nuke anything that deals with mysql or postgres, including all the setup junk.

**Comment by janxious** [comment]
ANd yeah, #29 is not necessary if we no longer connect to mysql or pg at all. :D

**Comment by janxious** [comment]
I tried a bunch of stuff to get the mysql bits to run correctly. There is apparently a big snafu around mysql on travis you can see at https://github.com/travis-ci/travis-ci/issues/6842

I think removing mysql and postgresql will probably be best, despite making it slightly harder to write tests for regressions in the future, should they arise.

**Comment by aried3r** [comment]
I ran into the same issues with MySQL on Travis.

I removed all MySQL and PostgreSQL code and am starting to see the first successful test runs on Travis. I'll have a look what makes the other ones fail.

**Comment by aried3r** [comment]
A green test run!

How do you like the approach so far?

**Comment by aried3r** [comment]
Btw, I'd love to add the Rails 5.1 beta, but [allowing it to fail](https://docs.travis-ci.com/user/customizing-the-build/#Rows-that-are-Allowed-to-Fail) if you're ok with that.

Other TODOs include changing the README and adapting the CHANGELOG.

Also, I'd love to add RuboCop with a few selected rules, like forcing Ruby 1.9 hash syntax etc. But in another PR.

File: test/test_helper.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def setup_database_cleaner
 
 def create_test_tables
   schema_file = File.dirname(__FILE__) + "/schema.rb"
-  ["pg", "mysql", "sqlite"].each do |db|
+  ["sqlite"].each do |db|
```
Comment (janxious) [inline test/test_helper.rb]: This probably doesn't need to be inside an `each` block any more.

**Comment by janxious** [comment]
> How do you like the approach so far?

Seems correct. There are some instances of sqlite being mentioned specifically (in at least the transaction test), which is unneccessary.

> Btw, I'd love to add the Rails 5.1 beta, but allowing it to fail if you're ok with that.

👌 sounds fine

> Other TODOs include changing the README and adapting the CHANGELOG.

Not sure what you mean about the changelog. It probably doesn't need to change at all until a new version is cut. Even then there shouldn't be any changes to old entries aside from grammatical errors. Is that what you have in mind, or were you thinking something else?

• • •

What is up with this commit? https://github.com/expectedbehavior/acts_as_archival/pull/28/commits/c4f33ee227dd303e1fc9b59139ed2f31fa6f08dd 😁

**Comment by aried3r** [comment]
> Seems correct. There are some instances of sqlite being mentioned specifically (in at least the transaction test), which is unneccessary.

Done.

> Not sure what you mean about the changelog. It probably doesn't need to change at all until a new version is cut.

You're right! I guess only the README has to be adapted that this gem is no longer being tested explicitly on PostgreSQL and MySQL?

> What is up with this commit? c4f33ee 😁

Heh, I think it's because Ruby 2.4 has [unified Fixnum and Binum into Integer](http://blog.bigbinary.com/2016/11/18/ruby-2-4-unifies-fixnum-and-bignum-into-integer.html) which arel 5 (which I guess is what shipped with Rails 4.1) doesn't support. You can see [a failed test run here](https://travis-ci.org/expectedbehavior/acts_as_archival/jobs/211382029).

**Comment by aried3r** [comment]
Hmm, I'm not sure why Travis suddenly can't find Ruby 2.4 anymore in some cases. That's the _one_ thing that used to work, haha.

**Comment by aried3r** [comment]
I found https://github.com/travis-ci/travis-ci/issues/7077 and will add the minor versions as well to see if this clears things up.

Btw, sorry for the many messages, but one more thing. Do you want me to clean up the commits (squash a few of them together) or are you just gonna squash them all into one using GitHub's Squash+Merge?

**Comment by janxious** [comment]
> Do you want me to clean up the commits (squash a few of them together) or are you just gonna squash them all into one using GitHub's Squash+Merge?

If you prefer that. It's your PR. I don't normally squash because I prefer the full history.

Just let me know when you are done. I think this is 👍

**Comment by aried3r** [comment]
I'm done. I'm just waiting for another successful test run and it's ready to be merged from my point of view. Thanks for all the support!

**Comment by janxious** [comment]
👍 you are very welcome. Thanks for getting this green!

I added you to the contributor list. If you would like to not be there, let me know.

**Comment by janxious** [comment]
@aried3r 

>  It probably doesn't need to change at all until a new version is cut

Released 1.2.0 today. Thanks again for all the help and the push to clean things up!

**Comment by aried3r** [comment]
You're welcome! Thanks for the fast feedback loops.

**Pull request merged**